src/metpy/plots/declarative.py

```diff
@@ -1040,13 +1100,22 @@ def griddata(self):
     def plotdata(self):
         """Return the data for plotting.
 
-        The data array, x coordinates, and y coordinates.
+        The two dimension coordinates and the data array.
 
         """
-        x = self.griddata.metpy.x
-        y = self.griddata.metpy.y
-
-        return x, y, self.griddata
+        if (set(self.griddata.dims) == {'x', 'y'}
```

Comment:
Is there a reason not to do:
```python
try:
    x = self.griddata.metpy.x
    y = self.griddata.metpy.y
except AttributeError:
    x = self.griddata[self.griddata.dims[1]]
    y = self.griddata[self.griddata.dims[0]]

return x, y, self.griddata
```
This would seem less likely to break anything (i.e. things that have an x/y coordinates without calling them lat/lon or x/y).

Reply:
The code you suggested does work, but again, I think we are attempting to future proof here to make it easier to add different types of plots (and those some `elif` statements to handle other cases. @jthielen again might offer his own insight on this one too.

Reply:
@dopplershift We unfortunately can't do that since `x` and/or `y` from the accessor may still be present even when we are not in yx (or xy) space (likewise to the previous comment). But, yes, the implementation here is way to likely to break something (I honestly don't know why I even did it this way in the first place).

Like the other case, we'll probably need to leverage a try/except with `find_axis_number`:

```python
try:
    plot_x_dim = self.griddata.metpy.find_axis_number('x')
    plot_y_dim = self.griddata.metpy.find_axis_number('y')
except ValueError:
   plot_x_dim = 1
   plot_y_dim = 0

return (
    self.griddata[self.griddata.dims[plot_x_dim]],
    self.griddata[self.griddata.dims[plot_y_dim]],
    self.griddata
)
```

Reply:
I'm much happier with the implementation suggested by @jthielen here and below.